src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -127,7 +127,7 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.extra('encryption_key_vault', help='The Uri of the KeyVault')
         c.extra('encryption_key_version', help='The version of the KeyVault key')
         c.argument('encryption_key_source',
-                   arg_type=get_enum_type(['Microsoft.Storage', 'Microsoft.Keyvault'], 'Microsoft.Storage'),
+                   arg_type=get_enum_type(['Microsoft.Storage', 'Microsoft.Keyvault']),
```

Comment:
Intentionally removing the default here?

Reply:
Yes, having the default for the parameter will overwrite the previous value when sent along the service. This is part of the issue being solved.